### PR TITLE
Checkstyle fixes

### PR DESCRIPTION
**Pull request by mcaandewiel**

### CODE REVIEW

File: src/main/java/com/sem/btrouble/BTrouble.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
     private static HashMap<String, Boolean> preferences;
     private AbstractGame game;
+    @SuppressWarnings("unused")
```
Comment (hungnguyen94) [inline src/main/java/com/sem/btrouble/BTrouble.java:26]: If it isn't used, you might as well remove it.

File: src/main/java/com/sem/btrouble/game/AbstractGame.java [hunk 2]
```diff
@@ -25,7 +22,6 @@
     private LevelObserver view;
     private Timer levelTimer;
 
-    // Max time in seconds.
```
Comment (hungnguyen94) [inline src/main/java/com/sem/btrouble/game/AbstractGame.java:28]: This doesn't count as inline comments(I think). Although the name already makes it clear what it's use is.

File: src/main/java/com/sem/btrouble/BTrouble.java (hunk 1 above)
Comment (mcaandewiel) [inline src/main/java/com/sem/btrouble/BTrouble.java:26]: This is an Eclipse problem. Sometimes when you initialize a dataset in your constructor, it shows up as unused.

File: src/main/java/com/sem/btrouble/model/GamePowerUp.java [hunk 3]
```diff
@@ -14,6 +15,7 @@
  * @author Martin
  *
  */
+@SuppressWarnings("serial")
```
Comment (hungnguyen94) [inline src/main/java/com/sem/btrouble/model/GamePowerUp.java:18]: I don't get serial warnings in IntelliJ. Maybe it is Eclipse specific? 
Though, we could fix this by not extending these classes from the Slick2D Shape class, which implements Serializable(which is the cause of this warning).

File: src/main/java/com/sem/btrouble/model/PowerUpFactory.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
  * @author Martin
  *
  */
-public final class PowerUpFactory {
+public abstract class PowerUpFactory {
```
Comment (hungnguyen94) [inline src/main/java/com/sem/btrouble/model/PowerUpFactory.java:10]: I don't think this class should be abstract.

File: src/main/java/com/sem/btrouble/model/SlowPowerUp.java [hunk 5]
```diff
@@ -33,12 +33,18 @@ public SlowPowerUp() {
...
+     * @param expirationTime after which the powerup expires.
      */
     public SlowPowerUp(float xpos, float ypos, int expirationTime) {
         super(xpos, ypos, expirationTime);
         on = false;
     }
     
+    @Override
+    public int hashCode() {
+      return 1;
```
Comment (hungnguyen94) [inline src/main/java/com/sem/btrouble/model/SlowPowerUp.java]: If you only return a fixed value, you break the use of any data structure that uses a hash. 
The generated hashCode function by your IDE is usually good enough, so I recommend using that.

File: src/main/java/com/sem/btrouble/model/Wallet.java [hunk 6]
```diff
@@ -35,6 +35,11 @@ public int getValue() {
         return value;
     }
 
+    @Override
+    public int hashCode() {
+      return 1;
```
Comment (hungnguyen94) [inline src/main/java/com/sem/btrouble/model/Wallet.java]: I recommend not returning fixed values for hashCode functions.

File: src/test/java/com/sem/btrouble/BubbleCollisionTests.java [hunk 7]
```diff
@@ -11,29 +11,33 @@
...
-    private static final float X = 0;
-    private static final float Y = 0;
     private static final float CENTER_X = 9f;
     private static final float CENTER_Y = 11f;
     private static final float HEIGHT = 14f;
     private static final float WIDTH = 15f;
 
+    /**
+     * Setup a bubble at the specified location.
+     */
```
Comment (hungnguyen94) [inline src/test/java/com/sem/btrouble/BubbleCollisionTests.java:32]: I thought Checkstyle was disabled for files in the Test directory?

**Comment by hungnguyen94** [comment]
Looks good. 
I do recommend implementing/generating the hashCode functions, as the CollisionHandler uses a HashSet to store objects. Returning a fixed value might break it.

File: src/main/java/com/sem/btrouble/game/AbstractGame.java (hunk 2 above)
Comment (rubenwiersma) [inline src/main/java/com/sem/btrouble/game/AbstractGame.java:28]: It's an inline comment

**Comment by ccberg** [comment]
Jawel, pilsch ligt in de vriezer

**Pull request closed without merging**